src/main/scala/pipedsl/codegen/bsv/BluespecGeneration.scala

```diff
@@ -431,7 +432,7 @@ object BluespecGeneration {
           vars + (m.name -> nvar)
         case _ => vars
       }
-    })
+    }) + (mod.name -> BVar("_lock_" + mod.name, translator.getLockedModType(LockImplementation.getModLockImpl)))
```

Comment:
This is manually adding a single global lock for the module - not sure what this is for